### ISSUE TITLE
Change order of type variables in `hrecField` and `pfield`

### ISSUE THREAD

**Issue by MangoIV** [Plutonomicon/plutarch-plutus#397]
### Problem

Using `hrecField` and `pfield` would be much nicer with the output polymorphism if we could do something like:
```haskell
pfield @"bla" @PSomething 
-- or 
pfield @"bla" @(PAsData PSomething) 
-- instead of 
pfield @"bla" @_ @_ @_ @_ @_ @PSomething
```
Same goes for hrecField. 

### proposed solution

In particular an ordering like `label`, `origin type`, `result type` would be good. 

### relevant issue 

#275

**Comment by TotallyNotChase** [comment]
agreed, this would make the type inference bugs less annoying - though I do still hope we can get to a real solution for the type inference.

**Comment by MangoIV** [comment]
Yeah, this is not only for the linked issue but also for situations like this:  
```haskell
pfield @"bla" x #== pfield @"blup" y
```
In this case the typechecker should complain if the type you wanna compare by is not specified.

**Comment by MangoIV** [comment]
This has been resolved by #515
